pkg/api/controller.go

```diff
@@ -3635,6 +3636,57 @@ func (c *Controller) GetLakeFSVersion(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+func (c *Controller) ReportUsageEvent(w http.ResponseWriter, r *http.Request, body ReportUsageEventJSONRequestBody) {
+	ctx := r.Context()
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, ErrAuthenticatingRequest)
+		return
+	}
+
+	usageClass, usageName, usageCount := body.Class, body.Name, body.Count
+	if usageClass == "" {
+		writeError(w, r, http.StatusBadRequest, "invalid usage class: cannot use an empty string")
+		return
+	}
+
+	if usageName == "" {
+		writeError(w, r, http.StatusBadRequest, "invalid usage name: cannot use an empty string")
+		return
+	}
+
+	if usageCount <= 0 {
```

Comment:
```suggestion
	if usageCount < 0 {
```
In some cases it might be easier to allow 0... despite it lowering performance :-/

Reply:
Sure.  Say you add a feature to find all Deltas in my repo and you can do it through the existing API.  It makes sense for the GUI to report _how many_ Deltas it found.  Why should the API make you check that you found any Deltas before you're allowed to report how many you've found?